refactor(log): use writeable options class to update config file

calahatras opened the pull request:
move option for spaces to view options in log page
use converter instead of separate flags for file view options in log page
GitDiffControl reacts to changes to ShowSpacesAsDots and rerenders file diff if selected diff is text changes

calahatras committed:
refactor(log): move option for spaces to view options, use converter instead of separate flags for file view options

mickedefault commented on GitOut/Features/Git/Log/GitLogPage.xaml:
```diff
@@ -425,9 +442,25 @@
-                            <Border Style="{StaticResource CardPanelStyle}" MinWidth="180" Margin="16 8" Padding="0">
+                            <Grid.Resources>
+                                <Style x:Key="PopoverCheckboxListItemStyle" TargetType="{x:Type CheckBox}" BasedOn="{StaticResource MaterialCheckBoxStyle}">
+                                    <Setter  Property="Margin" Value="0 8" />
```
adjust the two spaces to one

calahatras replied:
Fixed the spaces and also updated the spacing for self-closing elements to be the same (without space)

calahatras committed:
refactor(log): remove double space and use same self-closing convention using one space

calahatras committed:
enhancement(options): add writable options and make log view model persist changes to show spaces as dots

mickedefault commented on GitOut/Features/Options/OptionsWriter.cs:
```diff
@@ -0,0 +1,31 @@
+?using System;
```
is.. is that a BOM?

calahatras replied:
da freq?

calahatras replied:
:fire: faahr

calahatras committed:
enhancement(stage): stage listens to changes for show spaces as dots

GitDiffControl reacts to changes to ShowSpacesAsDots and rerenders file diff if selected diff is text changes

mickedefault reviewed (approved):
:star:
